### ISSUE TITLE
Rule proposal: `no-negated-some-every`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1704]
### Description

I believe negative all `Array.every()` can write in positive `Array.some()`

```js
[
    ![].every(() => true) === [].some(() => !true),
    ![0].every(() => true) === [0].some(() => !true),
    ![1].every(() => true) === [1].some(() => !true),
    ![].every(() => false) === [].some(() => !false),
    ![0].every(() => false) === [0].some(() => !false),
    ![1].every(() => false) === [1].some(() => !false),
]
````

### Fail

```js
if (!array.some(element => test(element)));
```

```js
if (!array.every(element => test(element)));
```

### Pass

```js
if (array.every(element => !test(element)));
```

```js
if (array.some(element => !test(element)));
```

```js
if (!array.every(Boolean));
```

**Comment by fisker** [comment]
I remember we have a rule proposal to prevent unnecessary negated condition, but I forget what's in that proposal and I can't find it.

**Comment by sindresorhus** [comment]
> I remember we have a rule proposal to prevent unnecessary negated condition, but I forget what's in that proposal and I can't find it.

https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1073 ?

**Comment by fisker** [comment]
Not this one, I'll search issues again.

**Comment by fisker** [comment]
This #856. Kind of related, but this is more complicated.

**Comment by papb** [comment]
> I believe all negative `Array.every()` can write in positive `Array.some()`

They can, but I think it might worsen readability sometimes.

**Comment by sindresorhus** [comment]
> They can, but I think it might worsen readability sometimes.

Example?

**Comment by sindresorhus** [comment]
Accepted
